aio_pika/robust_connection.py

```diff
@@ -72,13 +72,13 @@ def add_reconnect_callback(self, callback: Callable[[], None]):
         self._on_reconnect_callbacks.add(callback)
 
     async def connect(self, timeout=None):
-        if self.fail_fast:
-            return await super().connect(timeout=timeout)
-
         while True:
             try:
-                await super().connect(timeout=timeout)
-            except (ConnectionError, AMQPError):
+                return await super().connect(timeout=timeout)
+            except:
```

Comment:
@mosquito I use broad exception here because there are `pamqp` exceptions can be raised for instance when connecting while RabbitMQ is starting. Unfortunately `pamqp.specification` exceptions aren't inherited from base `PAMQPException` so we should do something like this if broad exception is undesirable:
```python
except (
    ConnectionError,
    AMQPError,
    pamqp.exceptions.PAMQPException
) + tuple(pamqp.specification.ERRORS.values()):
```